pkg/apis/intevents/v1beta1/brokercell_types.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+)
+
+// +genclient
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BrokerCell manages the set of data plane components servicing
+// one or more Broker objects and their associated Triggers.
+type BrokerCell struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the BrokerCell.
+	Spec BrokerCellSpec `json:"spec,omitempty"`
+
+	// Status represents the current state of the BrokerCell. This data may be out of
+	// date.
+	// +optional
+	Status BrokerCellStatus `json:"status,omitempty"`
+}
+
+var (
+	// Check that BrokerCell can be validated, can be defaulted, and has immutable fields.
+	_ apis.Validatable = (*BrokerCell)(nil)
+	_ apis.Defaultable = (*BrokerCell)(nil)
+
+	// Check that BrokerCell can return its spec untyped.
+	_ apis.HasSpec = (*BrokerCell)(nil)
+
+	_ runtime.Object = (*BrokerCell)(nil)
+
+	// Check that we can create OwnerReferences to a BrokerCell.
+	_ kmeta.OwnerRefable = (*BrokerCell)(nil)
+)
+
+// BrokerCellSpec defines the desired state of a Brokercell.
+type BrokerCellSpec struct {
+}
+
+// BrokerCellStatus represents the current state of a BrokerCell.
+type BrokerCellStatus struct {
+	// inherits duck/v1 Status, which currently provides:
+	// * ObservedGeneration - the 'Generation' of the Service that was last processed by the controller.
+	// * Conditions - the latest available observations of a resource's current state.
+	duckv1.Status `json:",inline"`
+
+	// IngressTemplate contains a URI template used to generate Broker
+	// ingress addresses. It may contain variables `name` and `namespace`.
```

Comment:
Will those variables be wrapped with anything? E.g. `$name` or `<name>`.

Maybe just add an example, which will make it clear.

Reply:
They will be wrapped in `{}` as defined in [RFC6570](https://tools.ietf.org/html/rfc6570). I'll add an example.